python3Packages.py-multihash: 1.0.0 -> 2.0.1

Luflosi opened the pull request:
###### Motivation for this change
https://github.com/multiformats/py-multihash/releases/tag/v2.0.1
After marking it as not broken in #112410, I'm now updating it to the latest version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Luflosi committed:
python3Packages.py-multihash: 1.0.0 -> 2.0.1

https://github.com/multiformats/py-multihash/releases/tag/v2.0.1

The `variants` package is not in the list of requirements for py-multihash, so remove it.
Switch to `pytestCheckHook` for the tests instead of using `pytest`.
Also slightly change the Python version requirements check.

fabaff commented:
Result of `nixpkgs-review pr 112433` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.py-multihash</li>
    <li>python38Packages.py-multihash</li>
    <li>python39Packages.py-multihash</li>
  </ul>
</details>

fabaff reviewed (commented):
- `variants` seems no longer be a [requirement](https://github.com/multiformats/py-multihash/blob/master/setup.py#L14-L18) 
- Please check if it's possible to switch to `pytestCheckHook`
- Please use the new naming convention
  - `pytestcov` -> `pytest-cov`
  - `pytestrunner`-> `pytest-runner`

r-rmcgibbo commented:
Result of `nixpkgs-review pr 112433` at a8c0dd59 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/26e65c7ded7c71eb2af64181eed3f5ee">python37Packages.py-multihash</a></li></li>
    <li><a href="https://gist.github.com/bbbd8040d9dfc87c84aad0e2b8864d42">python38Packages.py-multihash</a></li></li>
    <li><a href="https://gist.github.com/818610ce39b23b194eb26d204296a856">python39Packages.py-multihash</a></li></li>
  </ul>
</details>

Luflosi commented:
What's `pytest-runner` or `pytestrunner` for? `py-multihash` seems to build fine without this package. Can I safely remove it?

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112433` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.py-multihash</li>
<li>python38Packages.py-multihash</li>
<li>python39Packages.py-multihash</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112433` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.py-multihash</li>
<li>python38Packages.py-multihash</li>
<li>python39Packages.py-multihash</li>
</ul>
</details>

SuperSandro2000 commented:
> What's `pytest-runner` or `pytestrunner` for? `py-multihash` seems to build fine without this package. Can I safely remove it?

Yes, it is deprecated anyway and patched out of most packages.

Luflosi commented:
While removing the dependency on `pytest-runner` worked fine on macOS, it did not work on Linux, so I added it back in.
@fabaff Please double-check that I implemented everything you suggested correctly.
